src/bnet_client.py

```diff
@@ -0,0 +1,434 @@
+        response.ParseFromString(body)
+        log.debug(f"fetched connection:disconnect_notification token: {header.token} body: {response}")
+
+        # error_code: 3014
+
+        await self.disconnect()
+
+    async def _on_connection__connect_response(self, header, body):
+        response = connection_service_pb2.ConnectResponse()
+        response.ParseFromString(body)
+        # log.debug(f"fetched connection:connect_response token: {header.token} body: {response}")
+
+        self._imported_services_map.update(zip(self._imported_services, response.bind_response.imported_service_id))
+
+        await self.logon()
+
+    async def _watch_receive_message(self):
+        while True:
+            try:
+                await self._receive_message()
+            except Exception as e:
+                log.error(f'battle.net socket error: {repr(e)}')
+
+    async def _receive_message(self):
+        if self.reader is None:
+            raise BackendError("connection to battle.net is closed")
+
+        header_len_buf = await self.reader.read(2)
+        if not header_len_buf:
+            return
+        if len(header_len_buf) < 2:
+            raise BackendError("not enough data to read header length")
+        header_len = struct.unpack("!H", header_len_buf)[0]
+        header_buf = await self.reader.read(header_len)
+        if len(header_buf) < header_len:
+            raise BackendError("not enough data to read header data")
+
+        header = rpc_pb2.Header()
+        header.ParseFromString(header_buf)
+
+        body_len = header.size
+        if body_len:
+            body = await self.reader.read(body_len)
+            if len(body) < body_len:
+                raise BackendError("not enough data to read body data")
+        else:
+            body = None
+
+        log.debug(f"protobuf receiving data: service {header.service_id} method {header.method_id} token {header.token} status {header.status} size {header.size}")
+
+        # this is a response to a request from us
+        if header.service_id == self._RESPONSE_SERVICE_ID:
+            if header.token in self._token_callbacks:
+                callback = self._token_callbacks[header.token]
+                await callback(header, body)
+                del self._token_callbacks[header.token]
+            else:
+                log.debug("unexpected response received", str(header))
+
+        # server requesting something from us
+        try:
+            callback = self._exported_services_map[header.service_id][header.method_id]
+            await callback(header, body)
+        except Exception as e:
+            log.debug("failed to run service for received header", str(header), e)
+
+    async def logon(self):
+        request = authentication_service_pb2.LogonRequest()
+        # request.program = "App"
+        request.program = "BSAp"  # login in as mobile
+        request.platform = "Win"
+        request.locale = "enUS"
```

Comment:
for now we have no regional info fetched from proto communication so there is no need to specify that right?

Reply:
I tried not sending locale, but then get immediately disconnected. 
Do we have the correct value from authentication that we could use? I think we only have "region" ('eu', 'us', ...) so far, right?